extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js

```diff
@@ -102,6 +104,44 @@ const fontsToLoad = [
   },
 ];
 
+/**
+ * Installs the host CSS.
+ *
+ * We define this function instead of using runtime style install solutions for
+ * different reasons:
+ *
+ * 1. We don't do it as part of `registerElement` since components with
+ *    different names extend from the same abstract AmpStoryInteractive. As such,
+ *    we'd unnnecessarily install the same styles for every inheriting component
+ *    present on the page.
+ *
+ * 2. We don't use `installStylesForDoc()` since it's unnecessarily large. We
+ *    don't want to bundle it as part of this extension.
+ * @param {string} id
+ * @param {!Element|!ShadowRoot} cssRoot
+ * @param {!Element} sibling
+ * @param {string} cssText
+ */
+function installHostCssOnce(id, cssRoot, sibling, cssText) {
+  if (cssRoot.querySelector(`#${escapeCssSelectorIdent(id)}`)) {
```

Comment:
I think it would work well if we kept the system `style[amp-extension="amp-story-interactive"]` to install the service, why are we doing this instead?

Reply:
There are two options to use the existing `style[amp-extension="amp-story-interactive"]` system:

1. By bundling `installExtensionsForDoc()` as part of `amp-story-interactive`, which has a `~0.16 kb` impact on `amp-story-interactive-1.0.js`.

2. By placing the host CSS in every inherited component, which has a variable impact on runtime performance by overriding styles once for every tag name defined in the extension (10x).

I think the size tradeoff `amp-story-interactive-1.0.js` is more reasonable, so I chose to write a function to install once. However, the hand-rolled function is barely smaller than `installExtensionsForDoc()`, so I removed it and used the latter instead.